### PR TITLE
r/aws_ses_event_destination: Refresh resource state in the Read method

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/13464.
Relates https://github.com/terraform-providers/terraform-provider-aws/pull/13344.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_ses_event_destination: Correctly refresh resource state
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSESEventDestination_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSSESEventDestination_ -timeout 120m
=== RUN   TestAccAWSSESEventDestination_basic
=== PAUSE TestAccAWSSESEventDestination_basic
=== RUN   TestAccAWSSESEventDestination_disappears
=== PAUSE TestAccAWSSESEventDestination_disappears
=== CONT  TestAccAWSSESEventDestination_basic
=== CONT  TestAccAWSSESEventDestination_disappears
--- PASS: TestAccAWSSESEventDestination_disappears (120.14s)
--- PASS: TestAccAWSSESEventDestination_basic (122.08s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	122.122s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
```console
$ grep -Pzo "Read\(d\s+\*schema\.ResourceData,\s+meta\s+interface\{\}\)\s+error\s+{\s+return\s+nil\s+" aws/*.go
aws/resource_aws_ses_event_destination.go:Read(d *schema.ResourceData, meta interface{}) error {

	return nil
```

`aws_ses_event_destination` seems to be the only resource which has this behavior.

File: aws/resource_aws_ses_event_destination_test.go [hunk 1]
```diff
@@ -4,24 +4,21 @@ import (
...
 	"testing"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ses"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
 func TestAccAWSSESEventDestination_basic(t *testing.T) {
```
Comment (bflad) [inline aws/resource_aws_ses_event_destination_test.go:14]: It appears this resource notes import support, however the import testing is not present, which would have caught this issue as well:

```
=== CONT  TestAccAWSSESEventDestination_basic
    TestAccAWSSESEventDestination_basic: testing.go:684: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.

        (map[string]string) {
        }


        (map[string]string) (len=13) {
         (string) (len=24) "cloudwatch_destination.#": (string) (len=1) "2",
         (string) (len=47) "cloudwatch_destination.3035139398.default_value": (string) (len=7) "default",
         (string) (len=48) "cloudwatch_destination.3035139398.dimension_name": (string) (len=13) "ses:source-ip",
         (string) (len=46) "cloudwatch_destination.3035139398.value_source": (string) (len=10) "messageTag",
         (string) (len=46) "cloudwatch_destination.996466118.default_value": (string) (len=7) "default",
         (string) (len=47) "cloudwatch_destination.996466118.dimension_name": (string) (len=9) "dimension",
         (string) (len=45) "cloudwatch_destination.996466118.value_source": (string) (len=11) "emailHeader",
         (string) (len=22) "configuration_set_name": (string) (len=33) "tf_acc_cfg_ses_event_dst_stnw0jtp",
         (string) (len=7) "enabled": (string) (len=4) "true",
         (string) (len=16) "matching_types.#": (string) (len=1) "2",
         (string) (len=25) "matching_types.2185032793": (string) (len=6) "bounce",
         (string) (len=25) "matching_types.2809346765": (string) (len=4) "send",
         (string) (len=4) "name": (string) (len=36) "tf_acc_event_dst_cloudwatch_stnw0jtp"
        }
```

Can you please add the `ImportStateVerify` testing for this?

```go
			{
				ResourceName:      cloudwatchDestinationResourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
			{
				ResourceName:      kinesisDestinationResourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
			{
				ResourceName:      snsDestinationResourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```

Comment (ewbankkit) [reply aws/resource_aws_ses_event_destination_test.go:14]: Will do.
Something more needs to be fixed:

```console
--- FAIL: TestAccAWSSESEventDestination_basic (123.80s)
    testing.go:684: Step 1 error: error reading SES Configuration Set Event Destination (tf-acc-test-9116975414862209134): InvalidParameterValue: The configuration set name must be specified.
        	status code: 400, request id: 1c004c98-a37f-46b8-b276-b5393289b3fe
```

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
n_ -timeout 120m
=== RUN   TestAccAWSSESEventDestination_basic
=== PAUSE TestAccAWSSESEventDestination_basic
=== RUN   TestAccAWSSESEventDestination_disappears
=== PAUSE TestAccAWSSESEventDestination_disappears
=== CONT  TestAccAWSSESEventDestination_basic
=== CONT  TestAccAWSSESEventDestination_disappears
--- PASS: TestAccAWSSESEventDestination_disappears (122.44s)
--- PASS: TestAccAWSSESEventDestination_basic (137.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	137.132s
```

File: aws/resource_aws_ses_event_destination_test.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_ses_event_destination_test.go:14]: Need to specify the configuration set name and event destination name when importing.

Comment (ewbankkit) [reply aws/resource_aws_ses_event_destination_test.go:14]: Done.

**Pull request merged**